core/modules/ccontrol/UserQuerySelect.cc

```diff
@@ -214,11 +214,13 @@ std::string UserQuerySelect::getResultQuery() const {
     auto const& valueExprList = *_qSession->getStmt().getSelectList().getValueExprList();
     for (auto const& valueExpr : valueExprList) {
         if (valueExpr->isStar()) {
-            std::string errMsg;
             auto useSelectList = std::make_shared<query::SelectList>();
             useSelectList->addValueExpr(valueExpr);
             query::SelectStmt starStmt(useSelectList, _qSession->getStmt().getFromList().clone());
-            auto schema = _infileMerger->getSchemaForQueryResults(starStmt, errMsg);
+            sql::Schema schema;
+            if (not _infileMerger->getSchemaForQueryResults(starStmt, schema)) {
+                _errorExtra = "Could not get schema for query results.";
```

Comment:
This message may be too cryptic if this is all that we pass to client, maybe some extra context would be useful if we want to make it easier for users and ourselves to debug this rare situation later.

Reply:
Also, don't you want to stop right away if `getSchemaForQueryResults()` failed? Is it defined what state `schema` is going to have in that case?